Added fix for window minimize bug

bentoBAUX opened the pull request:
Added a check for width or height == 0. If yes, the window is minimized and the current frame is skipped.

bentoBAUX committed:
Added fix for window minimize bug

Added a check for width or height == 0. If yes, current frame is skipped.

Ceyron commented:
Hi @bentoBAUX,

Thank you very much for the proposed fix!

Two points:
1. Do you have the chance to test your fix across different OS? On Zulip, your screenshot indicated a Windows system.
2. Could you revert the stylistic changes in the pull request? Probably, you have a code autoformatter. For consistency, it would be best to maintain the current style. If I see correctly, your fix is only lines 131 ff.

bentoBAUX committed:
Fixed code style

Removed the curly brackets to consistent with the other simple if-statement at line 126. I also moved the fix to the top of the function since it made no difference.

bentoBAUX commented:
Hello @Ceyron,

Thank you for the feedback! You raised points that I did not think about.

I took your advice to test it on Linux Ubuntu also and I found out that the problem only occurs on Windows 11, so the fix is for Windows only. Unfortunately, I am unable to test on a MacOS. I also attempted to fix the style of my code but I am not too sure if it was what you were looking for. Do let me know if it isn't correct and I will change it!

Ceyron commented:
> I took your advice to test it on Linux Ubuntu also and I found out that the problem only occurs on Windows 11, so the fix is for Windows only. Unfortunately, I am unable to test on a MacOS.

I will tag @wi-re, maybe he can comment on this or test it on MacOS.

> I also attempted to fix the style of my code but I am not too sure if it was what you were looking for. Do let me know if it isn't correct and I will change it!

Sorry for not being precise about this. If you click on the tab `Files changed` at the top of the page, you can see that your commit added new spaces and automatically hard-wrapped certain lines. I am pretty sure this is caused by an auto code-formatter you use. I would like it if you could push another commit to the same branch you are currently using that removes these changes, such that the net difference is only the fix at lines 131 ff that you introduced.

Thanks again for reporting this and proposing the fix! 😊

bentoBAUX commented:
Ahhh! I understand what you mean now. I will make the necessary changes once I'm back at home after my afternoon lectures. 😁

bentoBAUX committed:
Reverted Style Changes

I copied and pasted the original Renderer.cpp from main into this and just added my fix. The file should be identical to the main branch except for the fix now.

bentoBAUX commented:
So I have reverted the stylistic changes done by my auto-formatter. The file should be identical to the main branch except for the part where I added my fix. Do let me know if there's still something to be changed. 

Thank you so much for your patience! 😅

Ceyron commented:
Thanks for updating the style!

I changed the base to a development branch with updates made during this winter term to ensure that the main branch is not breaking. We will merge the changes once the winter term (and the programming exercises) are over. They will then be visible to the next Game Physics iteration in one year.

Let's wait for a review by Rene (@wi-re) since he is more familiar with GLFW than I am.

bentoBAUX commented:
Sounds good! It was an honour to contribute. 😁

Ceyron commented:
Hi @Davidxdydz,
do you have some time to look at this PR.
